### PR TITLE
KFLUXINFRA-799: Creating Cluster Secret Store for Insights

### PR DESCRIPTION
**Pull request by manish-jangra**
Creating Cluster Secret Store for Insights to retreive secrets in konflux build pipelines.

### CODE REVIEW

File: components/cluster-secret-store/production/insights-approle-id-patch.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+- op: add
+  path: /spec/provider/vault/auth/appRole/roleId
+  value: 573e6024-0c58-7e95-365b-d2ad6127e838
```
Comment (hugares) [inline components/cluster-secret-store/production/insights-approle-id-patch.yaml:4]: Is it intentional to have the same app-role in stating and production?

Comment (skateman) [reply components/cluster-secret-store/production/insights-approle-id-patch.yaml:4]: I'm not sure if the secret is needed in staging at all, but we don't have a separate staging one. I guess it's safe as we use the secrets only for RHEL subscription and it does not matter where the built containers get the subscription context from.

File: components/cluster-secret-store/base/insights-secret-store.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        # App Role auth mechanism
+        # https://www.vaultproject.io/docs/auth/approle
+        appRole:
+          # Path where the App Role authentication backend is mounted
+          path: approle
+          # RoleID configured in the App Role authentication backend
+          roleId: 
+          # Reference to a key in a K8 Secret that contains the App Role SecretId
+          secretRef:
+            name: insights-appsre-vault
```
Comment (gbenhaim) [inline components/cluster-secret-store/base/insights-secret-store.yaml:25]: can we deploy this secret from our vault? (not a blocker for this pr)

**Pull request merged**